docker/scripts/functions.sh

```diff
@@ -210,6 +210,82 @@ function retrieve_current_bios_config() {
 	fi
 }
 
+# usage: normalize_dell_bios_config_file $config_filename
+# strips out irrelevant config sections to prevent meaningless diffs
+function normalize_dell_bios_config_file() {
+	local config_file=$1
+	local config_file_normalized="${config_file}.normalized"
+
+	if [[ ! -f "${config_file}" ]]; then
+		echo "Error: missing BIOS config file [${config_file}] to normalize"
+		return 1
+	fi
+
+	# Delete irrelevant sections from the JSON config to normalize for diff'ing
+	jq 'del(.SystemConfiguration.ServiceTag) |
+		  del(.SystemConfiguration.TimeStamp) |
+			del(.SystemConfiguration.Components[] | select(.FQDD=="BIOS.Setup.1-1").Attributes[] | select(.Name=="SetBootOrderEn")) |
+			del(.SystemConfiguration.Components[] | select(.FQDD=="BIOS.Setup.1-1").Attributes[] | select(.Name=="BiosBootSeq")) |
+			del(.SystemConfiguration.Components[] | select(.FQDD=="iDRAC.Embedded.1").Attributes[] | select(.Name=="NIC.1#DNSRacName")) |
+			del(.SystemConfiguration.Components[] | select(.FQDD=="System.Embedded.1").Attributes[] | select(.Name=="ServerOS.1#HostName"))' \
+		"${config_file}" >"${config_file_normalized}"
+}
+
+# usage: normalize_supermicro_bios_config_file $config_filename
+# strips out irrelevant config sections to prevent meaningless diffs
+function normalize_supermicro_bios_config_file() {
+	local config_file=$1
+	local config_file_normalized="${config_file}.normalized"
+
+	if [[ ! -f "${config_file}" ]]; then
+		echo "Error: missing BIOS config file [${config_file}] to normalize"
+		return 1
+	fi
+
+	cp "${config_file}" "${config_file_normalized}"
+
+	# File generation timestamp
+	sed --in-place '/File generated at/d' "${config_file_normalized}"
+	# ME firmware status
+	sed --in-place '/ME Firmware Status/d' "${config_file_normalized}"
+	# RAM Topology
+	sed --in-place '/P[1|2] DIMM[[:alpha:]][[:digit:]]/d' "${config_file_normalized}"
+	# Hard drive serial numbers
+	sed --in-place '/HDD Serial Number/d' "${config_file_normalized}"
+	# PXE boot wait time
+	sed --in-place '/PXE boot wait time/d' "${config_file_normalized}"
+	# Boot mode select
+	sed --in-place '/Option ValidIf.*Boot mode select/d' "${config_file_normalized}"
+	# Boot option ordering
+	sed --in-place '/Boot Option.*selectedOption/d' "${config_file_normalized}"
+	# TCG storage hardware
+	sed --in-place '/Menu name.*TOSHIBA.*order/d' "${config_file_normalized}"
+}
+
+# usage: compare_bios_config_files $vendor $config_file
+function compare_bios_config_files() {
+	local vendor=$1
+	local config_file="bios-configs-latest/$2"
+	local config_file_normalized="${config_file}.normalized"
+	local current_config="current_bios.txt"
+	local current_config_normalized="${current_config}.normalized"
```

Comment:
vendor is unused right? Also why not just pass the normalized file paths to the func?

Reply:
Yep, vendor was an artifact from before I broke out the vendor-specific normalization code into its own functions. And I've specified the bios-configs-latest/ file path in the function call rather than inside the function.